main.py

```
@@ -148,213 +91,234 @@ async def send_message(self, connection_id: str, message: str):
+    return new_provider
+
+@app.get("/admin/providers", dependencies=[Depends(auth.get_current_admin_user)])
+async def list_providers(db: Session = Depends(get_db)):
+    return db.query(models.ProviderKey).all()
+
+@app.get("/admin/usage", dependencies=[Depends(auth.get_current_admin_user)])
+async def get_all_usage(db: Session = Depends(get_db)):
+    return db.query(models.UsageLog).order_by(models.UsageLog.created_at.desc()).limit(100).all()
+
+@app.get("/user/usage")
+async def get_user_usage(current_user: models.User = Depends(auth.get_current_active_user), db: Session = Depends(get_db)):
```

Ruff (0.15.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)